### ISSUE TITLE
[Bug] Tags rendering is glitchy

### ISSUE THREAD

**Issue by username0x0a** [ForkIssues/Tracker#1932]
Tags in the commit tree are rendered in a glitchy way, like if the drawing is constrained to the current scroll offset – when a branch or tag rect goes off the active area, its border and inner area are cropped, also the branch or tag name is sticky to the absolute scroll area.

<img width="328" alt="Screenshot 2023-07-13 at 21 01 49" src="https://github.com/fork-dev/Tracker/assets/5228484/21547eed-2367-459a-8389-e518bf97ccfe">

Attaching a screen recording to express the story itself.

Fork v2.30 on macOS 14 Sonoma Beta 3 but I guess older OSes are affected as well, I don't think this could be something related to `AppKit` updates.

https://github.com/fork-dev/Tracker/assets/5228484/faea70be-c397-4e97-8730-f9ba25b4cef2

**Comment by DanPristupov** [comment]
> I guess older OSes are affected as well

This functionality has been working almost untouched for a decade.

> I don't think this could be something related to AppKit updates.

What makes you think so?

P.S. Did you try to restart the computer?

**Comment by username0x0a** [comment]
Yep, fresh OS reboot. I've rechecked with Ventura & it works okay there. 🤔 So, apparently some change in drawing routine coming with Sonoma? Sorry for a (for now) false alarm, at first sight, it looked a bit like some `drawRect`-level issue that we'd hardly notice as we don't see the actual code, hopefully Apple guys will address/fix it, but I'd probably keep it around just as part of the Sonoma checklist. 💡

**Comment by username0x0a** [comment]
As of today, I've rechecked it with Fork 2.33 & Sonoma Beta 7.

I almost thought it was gone for a while, but later discovered the issue persists when scrolling UP the Commit list slowly. 😅

Hope it helps in case it's not fixed in AppKit. 👍

https://github.com/fork-dev/Tracker/assets/5228484/f9a2d4e8-fbb4-468d-a3d1-25e533939b82
